docs/docs/guides/creating-your-own-commands.md

```diff
@@ -0,0 +1,118 @@
+# Creating your own Commands
+
+Commands are used to interact with nodestream from the command line.
+
+## Creating your own command
+
+Commands are implemented as [cleo](https://cleo.readthedocs.io/en/latest/usage.html) commands via a subclass of `nodestream.cli.commands:NodestreamCommand`. 
+
+```python
+from nodestream.cli.commands import NodestreamCommand
+
+class MyCommand(NodestreamCommand):
+    name = "my-command"
+    description = "My command description"
+
+    async def handle_async(self) -> None:
+        self.line("Hello World!")
+```
+
+### Operations 
+
+Commands are typically executed as a series of operations. Operations are defined as seperate classes that from `nodestream.cli.operations:Operation`. 
+
+```python
+from nodestream.cli.command import NodestreamCommand
+from nodestream.cli.operations import Operation
+
+class MyOperation(Operation):
+    async def perform(self, command: NodestreamCommand):
+        command.line("Hello World!")
+```
+
+And then used as part of a command like so:
+
+```python
+from nodestream.cli.commands import NodestreamCommand
+
+class MyCommand(NodestreamCommand):
+    name = "my-command"
+    description = "My command description"
+
+    async def handle_async(self) -> None:
+        operation = MyOperation()
+        await self.run_operation(operation)
+```
+
+### Command Arguments
+
+Commands can take arguments and options from the command line. These are defined as class attributes on the command class. 
+
+```python
+from cleo.helpers import argument, option
+
+from nodestream.cli.commands import NodestreamCommand
+
+class MyCommand(NodestreamCommand):
+    name = "my-command"
+    description = "My command description"
+    arguments = [
+        argument("name", "the name of the project to create"),
+    ]
+
+    options = [
+        option("shout", "s", "Yell the message"),
+    ]
+
+    async def handle_async(self) -> None:
+        name = self.argument("name")
+        message = f"Hello {name}!"
+        if self.option("shout"):
+            message = message.upper()
+        self.line(message)
+```
+
+
+### Registering the Audit
```

Comment:
Should this say "Registering the Command"?

Reply:
Good catch - yes!